src/main/scala/com/wavesplatform/settings/BlockchainSettings.scala

```diff
@@ -156,10 +156,15 @@ object BlockchainSettings {
         val addressSchemeCharacter = config.as[String](s"$configPath.custom.address-scheme-character").charAt(0)
         val functionalitySettings = FunctionalitySettings.fromConfig(config)
         val genesisSettings = GenesisSettings.fromConfig(config)
-
         (addressSchemeCharacter, functionalitySettings, genesisSettings)
     }
 
-    BlockchainSettings(file, addressSchemeCharacter, functionalitySettings, genesisSettings)
+    BlockchainSettings(
+      blockchainFile = config.as[String](s"$configPath.blockchain-file"),
+      stateFile = config.as[String](s"$configPath.state-file"),
+      checkpointFile = config.as[String](s"$configPath.checkpoint-file"),
```

Comment:
Ficus supports custom value extractors, so how about creating a type class and change this to
```scala
blockchainFile = config.as[Path](s"$configPath.blockchain-file")
```